File: src/org/webfabric/web/servlet/ServletPath.java
package org.webfabric.web.servlet;

import org.webfabric.io.Path;

import javax.servlet.http.HttpServletRequest;

public class ServletPath implements Path {
    private final String value;

    public ServletPath(HttpServletRequest request) {
        this.value = request.getServletPath();
    }

    public ServletPath(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }
}

File: src/org/webfabric/web/servlet/ServletPath.scala
package org.webfabric.web.servlet;

import org.webfabric.io.Path;

import javax.servlet.http.HttpServletRequest;

class ServletPath(v: String) extends Path {
  def value(): String = v

  override def toString = v
}

object ServletPath {
  def apply(request: HttpServletRequest):ServletPath = {
    return new ServletPath(request.getServletPath());
  }
}